### PR TITLE
[LTC] Refactor class Shape

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request keeps track of refactoring of class Shape.
1. Remove Shape.IsTuple() from ShapeUtil.
2. Change Shape.at_element_type_ to Shape.scalar_type_.
3. Change Shape.dimensions_ to Shape.sizes_.
3. Change Shape.rank() to Shape.dim().
4. Change TsNode.shape_ to TsNode.shapes_.
5. Remove tuple shapes in class Shape.
4. Remove Shape.IsTuple().
5. A little bit more clean up on class Shape.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

To be noted:
It's easier to review the pull request one commit by another.

Depends on: #67755.

### CODE REVIEW

File: aten/src/ATen/templates/LazyIr.h [hunk 1]
```diff
@@ -10,22 +10,17 @@
 namespace torch_lazy_tensors {
 namespace ir {
 
-static lazy_tensors::Shape convertShape(
+static std::vector<lazy_tensors::Shape> convertShape(
```
Comment (wconstab) [inline aten/src/ATen/templates/LazyIr.h:13]: convertShapes? (s)

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/cat.cpp [hunk 2]
```diff
@@ -7,22 +7,17 @@ namespace torch_lazy_tensors {
 namespace ir {
 
 // TODO(whc) don't duplicate this
-static lazy_tensors::Shape convertShape(
+static std::vector<lazy_tensors::Shape> convertShape(
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/ops/cat.cpp:10]: should this be be moved into shape.h?

**Review by wconstab — APPROVED** [review]
Nice work, this is great stuff. Thanks!

File: aten/src/ATen/templates/LazyIr.h (hunk 1 above)
Comment (alanwaketan) [reply aten/src/ATen/templates/LazyIr.h:13]: Fixed.

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/cat.cpp (hunk 2 above)
Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/cat.cpp:10]: Yea, moved.

**Comment by alanwaketan** [comment]
@wconstab Thanks for approving this pull request.

**Pull request merged**